src/apps/applets/domain/constants.py

```diff
@@ -3,11 +3,24 @@
 
 class Role(str, Enum):
     ADMIN = "admin"
-    CONTENT_MANAGER = "content manager"
-    DATA_MANAGER = "data manager"
-    CASE_MANAGER = "case manager"
-    RESPONDENTS_MANAGER = "respondents manager"
-    REVIEWERS_MANAGER = "reviewers manager"
-    MANAGERS_MANAGER = "managers manager"
+    MANAGER = "manager"
+    COORDINATOR = "coordinator"
+    EDITOR = "editor"
     REVIEWER = "reviewer"
     RESPONDENT = "respondent"
+
+
+class RoleLevel(int, Enum):
+    ADMIN = 1  # the highest
+    MANAGER = 2
+    COORDINATOR = 3
+    EDITOR = 4
+    REVIEWER = 5
+    RESPONDENT = 6  # the lowest
+
+    @classmethod
+    def by_role(cls, role: Role):
+        """
+        returns level by role, if not found return the lowest role
+        """
+        return cls._member_map_.get(role.name, 1000)
```

Comment:
I think it is not a good idea to use integers for enums. If you wanna create something, like a priority you can do this:

```python

class Role(str, Enum):
    ADMIN = "admin"
    MANAGER = "managers"
    COORDINATOR = "coordinator"
    # ... 

    @classmethod
    @lru_cache
    def as_list(cls) -> list[Role]:
        """Nothing is going to change the enum class in a runtime.
        Then it is better to cache the result from the method as far
        as we do quite a lot of accesses to it
        in order to Roles priorities comparings.
        """
        return [i for i in cls]

    def __lt__(self, other):
        """In order to use ordering just from the class defined values without any
        kind of `priority` values the index of the cached list of elements is used.
        """
        data = self.as_list()
        return data.index(self) < data.index(other)
```

Reply:
Actually, if you will add the type annotation to the `by_role` function you will see that it can return Enum instance or the integer it is a bad idea to do :)

Reply:
Another thing I guess is that using some immense random value is not a great idea :)
We can avoid having this method in a general cause in my example they could be compared like `Role.MANAGER > Role.ADMIN`